CONTRIBUTING.asciidoc

```diff
@@ -0,0 +1,356 @@
+= Contributing to ipa
+
+=== Installation
+
+[source]
+----
+git clone https://github.com/SUSE/ipa.git
+cd ipa
+
+# Activate virtual Environment then install
+# ipa and dev dependences in editable mode
+pip install -e .[dev]
+----
+
+ipa is now installed in the active virtual environment in development
+mode.
+
+=== Dev Requirements
+
+* bumpversion
+* pip>=7.0.0
+
+=== Testing Requirements
+
+* coverage
+* flake8
+* mock
+* pytest-cov
+* vcrpy
+
+=== Versions & Releases
+
+*ipa* adheres to Semantic versioning; see http://semver.org/ for details.
+
+link:https://pypi.python.org/pypi/bumpversion/[bumpversion] is used
+for release version management, and is configured in `setup.cfg`:
+
+----
+bumpversion major|minor|patch
+git push && git push --tags
+----
+
+=== Running the tests
+
+All tests should pass and test coverage should remain above 90%.
+
+The tests and coverage can be run directly via Pytest.
+
+----
+pytest --cov=ipa
+----
+
+They can also be run with tox integration which runs flake8 as well:
+
+=== Testing with tox
+
+==== Requirements
+
+* tox
+* tox-pyenv (testing multiple Python envs)
+
+==== Tox installation
+
+pip install .[tox]
+
+==== Setup pyenv
+
+See the link:https://github.com/pyenv/pyenv#installation[pyenv doc] for more
+info.
+
+==== Run tox
+
+----
+tox (test all versions)
+
+# or
+tox -e py{version}
+tox -e py36
+----
+
+=== Code Style
+
+Source should pass flake8 and pep8 standards.
+
+----
+flake8 ipa
+----
+
+== ipa Source Overview
+
+=== CLI
+
+ ipa test
+
+The test subcommand invokes a test suite on an image in the chosen
+public cloud environment. The image id, tests and provider are required
+arguments. See below for options:
+
+----
+Example: ipa test --ssh-private-key /path/to/key -d SLES Azure test_image
+----
+
+ ipa results
+
+The results subcommand displays the test results information. The path
+to the results xml is required.
+
+----
+Example: ipa results -v
+----
+
+The default location for results files is ~/ipa/results/ and the files
+are encoded with the timestamp of execution. For example,
+the results for an image test in ec2 would be found at
+~/ipa/results/ec2/{imageId}/{instanceId}/{datetime}.results.
+
+----
+Example results directory:
+
+ec2/:
+  ami-43243232/:
+    i-3432r4324y3t2/:
+      {datetime}.results
+    i-432423j3j2432/:
+      {datetime}.results
+----
+
+ ipa list
+
+The list subcommand displays the available tests.
+
+----
+Examples: ipa list
+----
+
+The list subcommand will return a list of test files in the default test
+directories. The verbose option will return a list of all available
+tests in all test files.
+
+----
+Example output:
+
+$ ipa list -v
+
+test_broken::test_broken
+test_image::test_image
+test_sles::test_sles
+test_sles::test_sles_1
+test_sles::test_sles_2
+----
+
+=== API
```

Comment:
Another document, development of tests or on top of IPA is a distinctly different activity than contributing to the project.